src/parser.js

```diff
@@ -843,8 +844,10 @@
       ownAttributes = extend(ownAttributes,
         extend(getGlobalStylesForElement(element, svgUid), fabric.parseStyleAttribute(element)));
 
-      fontSize = (parentAttributes && parentAttributes.fontSize ) ||
-                   ownAttributes['font-size'] || fabric.Text.DEFAULT_SVG_FONT_SIZE;
+      fontSize = parentFontSize = parentAttributes.fontSize || fabric.Text.DEFAULT_SVG_FONT_SIZE;
+      if (ownAttributes['font-size']) {
+        fontSize = parseUnit(ownAttributes['font-size'], parentFontSize);
+      }
```

Comment:
@rikkarv the idea here is that 
- parseUnit should be used always.
- if the current fontSize is specified in EM ( i think is sort of legit ) we must reference the parent fontsize

Reply:
ok this is not good needs additional tweaking as per the uploaded test.